src/librustc/diagnostics.rs

```diff
@@ -1135,6 +1135,63 @@ fn main() {
 
 Rust only looks at the signature of the called function, as such it must
 already specify all requirements that will be used for every type parameter.
+
+Here's another erroneous code example:
+```compile_fail,E0277
+trait A {}
+
+struct B;
+
+impl A for B {}
+
+// Let's assume that you would like to return a trait like below:
+fn foo(b: B) -> A {
+    b // won't compile
+}
+```
+
+Here are two solutions to fix this issue:
+
+1. `Box` your returned value
+```rust
+# use std::boxed::Box;
+# use std::borrow::Borrow;
+# use std::fmt::Debug;
+#
+# trait A: Debug {}
+#
+# #[derive(Debug)]
+# struct B;
+#
+# impl A for B {}
+#
+fn foo(b: B) -> Box<A> {
+    Box::new(b) // The `Sized` trait bound has been removed.
```

Comment:
Please write "error: [the error message]" in here instead.

Reply:
yup, `rustc` compiles this sample.